### PR TITLE
Add support for no_std

### PR DESCRIPTION
**Pull request by thomcc**
Only really dodgy thing is the way `abort` is implemented, but I think it should be fine still.

This probably increases MSRV but IDK how much that matters (wasn't listed in CI).

EDIT: Also made various crates optional. (`serde` because compile times, `stable_deref_trait` because I don't think it's very widely used anymore).

### CODE REVIEW

**Comment by thomcc** [comment]
Oh, I guess https://crates.io/crates/servo_arc is on crates.io? Is this crate something people should use?

I guess I don't need all of it's functionality...

**Comment by Manishearth** [comment]
@thomcc i don't think the crates.io servo_arc is used, this is the community maintained version of that crate, not as geared for Firefox use cases.

File: lib.rs [hunk 1]
```diff
@@ -238,6 +245,18 @@ impl<T: ?Sized> Arc<T> {
     }
 }
 
+// `no_std`-compatible abort by forcing a panic while already panicing.
+fn abort() -> ! {
```
Comment (Manishearth) [inline lib.rs:249]: Can we conditionally cfg this; when we have std we use the regular abort?

Comment (thomcc) [reply lib.rs:249]: Sure.

**Comment by Manishearth** [comment]
Thanks!

**Pull request merged**